### DISCUSSION TITLE
Facing issues in parquet file reads from GCS using velox Backend.

### DISCUSSION THREAD

**Discussion by pratham76** [apache/gluten#7414] [Q&A]
I am trying to read a parquet file from GCS using service credentials file as per this [doc](https://github.com/apache/incubator-gluten/blob/main/docs/get-started/VeloxGCS.md#configuring-gcs-using-a-credential-file) and have added following configurations.

```
        "spark.hadoop.fs.gs.auth.type": "SERVICE_ACCOUNT_JSON_KEYFILE",
        "spark.hadoop.fs.gs.auth.service.account.json.keyfile": "/mnts/applications/key.json",
```
though Java based spark works using these credentials. Gluten hangs with the following statement.
`W20241005 09:53:12.233268   188 GCSFileSystem.cpp:303] Config::gcsCredentials is empty`

Please help if there is any workaround for this?

**Comment by majetideepak** [comment]
This was recently fixed in Velox. See https://github.com/facebookincubator/velox/pull/11086
Can you try with the latest Gluten version?

**Reply by majetideepak** [reply]
We need two configs for GCS in Velox.
```
hive.gcs.json-key-file-path=/mnts/applications/key.json
hive.gcs.endpoint=storage.googleapis.com
```

**Reply by majetideepak** [reply]
@pratham76 I checked the code and you will need this PR https://github.com/apache/incubator-gluten/pull/7421
Can you please check with this patch?
I see that the default gcs endpoint should work as expected.

**Reply by pratham76** [reply]
sure, will verify on a build with this change.
